lib/pbench/server/unpack_tarballs.py

```diff
@@ -100,34 +65,39 @@ def unpack_tarballs(self, min_size: float, max_size: float) -> Results:
         Returns:
             Results tuple containing the counts of Total and Successful tarballs.
         """
-        tarlist = [
-            tarball
-            for tarball in self.config.ARCHIVE.glob(f"*/{self.LINKSRC}/*.tar.xz")
-            if min_size <= Path(tarball).stat().st_size < max_size
-        ]
-
-        ntotal = nsuccess = 0
-        cache_m = CacheManager(self.config, self.logger)
-
-        for tarball in sorted(tarlist):
-            ntotal += 1
-
-            try:
+        datasets = self.sync.next(Operation.UNPACK)
+        tarlist: list[Target] = []
+        for d in datasets:
+            t = Metadata.getvalue(d, Metadata.TARBALL_PATH)
+            if t:
```

Comment:
How about
```
            if not t:
                continue
```
which would allow us to out-dent the rest of the lines in this block....

But, more to the point, can this ever happen (or is this some sort of paranoia trap)?  How can we have a dataset with no tarball?  (I note that we don't have a unit test case for this scenario.)